lib/backend/report.go

```diff
@@ -118,6 +118,12 @@ func NewReporter(cfg ReporterConfig) (*Reporter, error) {
 		topRequestsCache:    cache,
 		slowRangeLogLimiter: rate.NewLimiter(rate.Every(time.Minute), 12),
 	}
+
+	// metrics that need to be initialized for use later, this makes it easier on the consumer of these metrics (Grafana dashboards, for one)
+	readRequestsFailed.WithLabelValues(r.Component)
+	writeRequestsFailed.WithLabelValues(r.Component)
+	writeRequestsFailedPrecondition.WithLabelValues(r.Component)
```

Comment:
Why are these the only three metrics that should be initialized? Can we generate and store all the labeled metric objects that we need so we're sure we're not going to accidentally leave something uninitialized until some time later?

Reply:
The others tend to self-initialize early enough when Teleport starts up to not be a problem.  I could loop through all (minus requests) metrics and initialize with the component label.  I think I'd have to skip initialization of [requests](https://github.com/gravitational/teleport/blob/f756a16795d015f7f5ee52b30ae980e09c6679bf/lib/backend/report.go#L657) given the other dimensions are not known beforehand.  Would that make this too messy? or acceptable?  That's my main hesitation.

Reply:
`r.requests = requests.MustCurryWith(prometheus.Labels{teleport.ComponentLabel: r.Component})`?